nicegui/storage.py

```diff
@@ -149,10 +151,26 @@ def general(self) -> Dict:
         """General storage shared between all users that is persisted on the server (where NiceGUI is executed)."""
         return self._general
 
+    @property
+    def client(self) -> ObservableDict:
+        """Client storage that is persisted on the server (where NiceGUI is executed) on a per client
+        connection basis.
+
+        The data is lost when the client disconnects through reloading the page, closing the tab or
+        navigating away from the page. It can be used to store data that is only relevant for the current view such
+        as filter settings on a dashboard or in-page navigation. As the data is not persisted it also allows the
+        storage of data structures such as database connections, pandas tables, numpy arrays, user specific ML models
+        or other living objects that are not serializable to JSON.
+        """
+        client = context.get_client()
```

Comment:
I think we should add a safe guard to ensure the storage.client is not used with the [auto-index page](https://nicegui.io/documentation/section_pages_routing#auto-index_page). Probably the same as 

https://github.com/zauberzeug/nicegui/blob/7a115b0f34df35703d030da55cb85c417ecc5d4b/nicegui/storage.py#L161-L163

Reply:
Done, actually the get_slot triggered an exception as well but this way it more obvious to the user why.